tempesta_fw/t/unit/test_sched_http.c

```diff
@@ -0,0 +1,300 @@
+	test_req("GET http://google.com/foo/baz/ HTTP/1.1\r\n\r\n", NULL);
+
+	cleanup_cfg();
+	test_conn_release_all(sg1);
+	test_conn_release_all(sg2);
+	test_conn_release_all(sg3);
+	test_conn_release_all(sg4);
+	test_conn_release_all(sg5);
+	test_conn_release_all(sg6);
+	test_conn_release_all(sg7);
+	test_conn_release_all(sg8);
+	test_conn_release_all(sg9);
+	test_conn_release_all(sg10);
+	test_sg_release_all();
+}
+
+typedef struct {
+	char *rule_str;
+	char *good_req_str;
+	char *bad_req_str;
+} TestCase;
+
+TestCase test_cases[] = {
+	{
+		.rule_str = "sched_http_rules {\nmatch default uri eq /foo;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\n\r\n",
+		.bad_req_str = "GET http://natsys-lab.com/foo2 HTTP/1.1\r\n\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default uri prefix /foo;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo2 HTTP/1.1\r\n\r\n",
+		.bad_req_str = "GET http://natsys-lab.com/bar HTTP/1.1\r\n\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default host eq natsys-lab.com;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\n\r\n",
+		.bad_req_str = "GET http://natsys-lab2.com/foo HTTP/1.1\r\n\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default host prefix natsys-lab;\n}\n",
+		.good_req_str = "GET http://natsys-lab2.com/foo HTTP/1.1\r\n\r\n",
+		.bad_req_str = "GET http://google.com/foo HTTP/1.1\r\n\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default hdr_host eq natsys-lab.com;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nHost: natsys-lab.com\r\n",
+		.bad_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nHost: natsys-lab2.com\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default hdr_host prefix natsys-lab;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nHost: natsys-lab2.com\r\n",
+		.bad_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nHost: google.com\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default hdr_conn eq Keep-Alive;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nConnection: Keep-Alive\r\n",
+		.bad_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nConnection: close\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default hdr_conn prefix Keep;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nConnection: Keep-Alive\r\n",
+		.bad_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nConnection: close\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default hdr_raw eq User-Agent:Bot;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nUser-Agent:Bot\r\n",
+		.bad_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nUser-Agent:Tot\r\n",
+	},
+	{
+		.rule_str = "sched_http_rules {\nmatch default hdr_raw prefix User-Agent;\n}\n",
+		.good_req_str = "GET http://natsys-lab.com/foo HTTP/1.1\r\nUser-Agent:Bot\r\n",
```

Comment:
Field names are case-insensitive. Maybe it's worth to test mismatching character cases in rule and requests too?